### PR TITLE
Remove Bandwidth Completely

### PR DESCRIPTION
**Pull request by reggaemuffin**
This pr removes bandwidth and reserve ration calculations completely.

It keeps the apis for them for backwards compatibility, but serves dummy values.

It removes all calculations regarding bandwidth to speed up replay.

Unit tests ensure that all apis are working and no bugs are present.

Related: steemit/steem#3040 steemit/steem#3029 steemit/steem#3022 steemit/steem#3006 

RPC node available for testing: `https://testbed.reggaemuffin.me/`

```
# curl -s --data '{"jsonrpc":"2.0", "method":"witness_api.get_account_bandwidth", "params":{"account":"steemit","type":"forum"}, "id":1}' https://testbed.reggaemuffin.me/
{"jsonrpc":"2.0","result":{"bandwidth":{"id":1,"account":"steemit","type":"forum","average_bandwidth":0,"lifetime_bandwidth":0,"last_bandwidth_update":"2016-03-24T16:00:00"}},"id":1}

# curl -s --data '{"jsonrpc":"2.0", "method":"condenser_api.get_dynamic_global_properties", "params":[], "id":1}' https://testbed.reggaemuffin.me           
{"jsonrpc":"2.0","result":{"head_block_number":26659863,"head_block_id":"0196cc17c732330347d8c044744e17f8ec474520","time":"2018-10-09T15:17:42","current_witness":"timcliff","total_pow":514415,"num_pow_witnesses":172,"virtual_supply":"296311820.681 STEEM","current_supply":"278306627.364 STEEM","confidential_supply":"0.000 STEEM","current_sbd_supply":"15034336.420 SBD","confidential_sbd_supply":"0.000 SBD","total_vesting_fund_steem":"196749512.428 STEEM","total_vesting_shares":"397282198615.768214 VESTS","total_reward_fund_steem":"0.000 STEEM","total_reward_shares2":"0","pending_rewarded_vesting_shares":"463838940.053835 VESTS","pending_rewarded_vesting_steem":"227396.446 STEEM","sbd_interest_rate":0,"sbd_print_rate":10000,"maximum_block_size":65536,"current_aslot":26754354,"recent_slots_filled":"340282366920938463463374607431768211455","participation_count":128,"last_irreversible_block_num":26659843,"vote_power_reserve_rate":10,"delegation_return_period":432000,"reverse_auction_seconds":900,"sbd_stop_percent":1000,"sbd_start_percent":900,"average_block_size":0,"current_reserve_ratio":200000000,"max_virtual_bandwidth":"0"},"id":1}
```

### CODE REVIEW

**Comment by bobinson** [comment]
build is working fine. Not adding any comments as I am not familiar with the rc code changes.

File: doc/building.md [hunk 1]
```diff
@@ -231,6 +231,68 @@ e.g.:
 
 This will only build `steemd`.
 
+## Building on Windows with CSL and CLion
```
Comment (voorash) [inline doc/building.md]: Is that supposed to say "with WSL and CLion"

**Comment by voorash** [comment]
Other than my one minor comment it looks good to me but I am also not familiar with the RC stuff so not adding myself as an actual reviewer.

File: doc/building.md (hunk 1 above)
Comment (reggaemuffin) [reply doc/building.md]: Fixed it :)

File: libraries/plugins/witness/witness_plugin.cpp [hunk 2]
```diff
@@ -260,29 +265,6 @@ namespace detail {
 
    void witness_plugin_impl::on_pre_apply_transaction( const chain::transaction_notification& note )
    {
```
Comment (bobinson) [inline libraries/plugins/witness/witness_plugin.cpp]: The commented out block here can be safely removed as the code with commented out code is deployed on majority of the witness servers and witnesses are working without issues.

File: libraries/plugins/witness/witness_plugin.cpp [hunk 3]
```diff
@@ -318,158 +300,9 @@ namespace detail {
 
    void witness_plugin_impl::on_post_apply_block( const block_notification& note )
    { try {
-      /*
```
Comment (bobinson) [inline libraries/plugins/witness/witness_plugin.cpp]: This commented code blockcan also be removed as there are no references to it and the code will look cleaner.

Comment (reggaemuffin) [reply libraries/plugins/witness/witness_plugin.cpp]: That is a delete line, it is already deleted :)

**Comment by bobinson** [comment]
LGTM

**Comment by reggaemuffin** [comment]
Don't merge yet as https://github.com/steemit/steem/commit/03d1de1906315548141a750f8a75c8c668041389 seems to be a similar patch by steemit inc 🤷‍♂️ will track progress on both

**Comment by reggaemuffin** [comment]
closing as a different patch removes bandwidth completely

**Pull request closed without merging**